arangod/Cluster/ServerState.cpp

```diff
@@ -647,23 +669,30 @@ bool ServerState::registerAtAgencyPhase2(AgencyComm& comm) {
   TRI_ASSERT(!_id.empty() && !_myEndpoint.empty());
 
   while (!application_features::ApplicationServer::isStopping()) {
-    VPackBuilder builder;
-    try {
-      VPackObjectBuilder b(&builder);
-      builder.add("endpoint", VPackValue(_myEndpoint));
-      builder.add("advertisedEndpoint", VPackValue(_advertisedEndpoint));
-      builder.add("host", VPackValue(getHost()));
-      builder.add("version", VPackValue(rest::Version::getNumericServerVersion()));
-      builder.add("versionString", VPackValue(rest::Version::getServerVersion()));
-      builder.add("engine", VPackValue(EngineSelectorFeature::engineName()));
-    } catch (...) {
-      LOG_TOPIC("de625", FATAL, arangodb::Logger::CLUSTER) << "out of memory";
-      FATAL_ERROR_EXIT();
-    }
-
-    auto result = comm.setValue(currentServersRegisteredPref + _id, builder.slice(), 0.0);
+    std::vector<std::shared_ptr<VPackBuilder>> builders;
+
+    auto buildSetOp = [this, &builders](char const* url, VPackValue const& v) {
+      builders.emplace_back(new VPackBuilder);
+      builders.back()->add(v);
+      return AgencyOperation(currentServersRegisteredPref + _id + "/" + url,
+                             AgencyValueOperationType::SET, builders.back()->slice());
+    };
+
+    AgencyWriteTransaction trx(
+        {buildSetOp("endpoint", VPackValue(_myEndpoint)),
+         buildSetOp("advertisedEndpoint", VPackValue(_advertisedEndpoint)),
+         buildSetOp("host", VPackValue(getHost())),
+         buildSetOp("version", VPackValue(rest::Version::getNumericServerVersion())),
+         buildSetOp("versionString", VPackValue(rest::Version::getServerVersion())),
+         buildSetOp("engine", VPackValue(EngineSelectorFeature::engineName())),
+         AgencyOperation(currentServersRegisteredPref + _id + "/rebootId",
+                         AgencySimpleOperationType::INCREMENT_OP),
```

Comment:
An increment on a non existing key stores 1 in there, correct?

Reply:
Yes @neunhoef showed it to me, and I tried it out myself as well.

Reply:
Maybe I should note this in a comment somewhere around there.